### PR TITLE
Harden events against race conditions

### PR DESCRIPTION
**Pull request by brunom**

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/wpf?pullRequest=5722) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/wpf?pullRequest=5722) before we can accept your contribution.<br/><br/>:x: brunom [sign now](https://cla.dotnetfoundation.org/dotnet/wpf?pullRequest=5722)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/wpf?pullRequest=5722) it.</sub>

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/MediaContext.cs [hunk 1]
```diff
@@ -1849,7 +1846,7 @@ private object InputMarkerMessageHandler(object arg)
                         // (TimeManager gets its tick time from MediaContext's IClock implementation).
                         // In the case where we can't query QPC or aren't doing interlocked presents,
                         // this will be equal to the current time, which is a good enough approximation.
-                        Rendering(this.Dispatcher, new RenderingEventArgs(_timeManager.LastTickTime));
+                        Rendering?.Invoke(this.Dispatcher, new RenderingEventArgs(_timeManager.LastTickTime));
```
Comment (lindexi) [inline src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/MediaContext.cs:1849]: May I know when Render is null?

Comment (vishalmsft) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/MediaContext.cs:1849]: @brunom There is already a null check line#1842

Comment (brunom) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/MediaContext.cs:1849]: @vishalmsft There is but the variable may have gone back to null in the meantime.

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/MediaContext.cs [hunk 2]
```diff
@@ -996,10 +996,7 @@ private void NotifySyncModeStatus(int enabledResult)
                     if (Channel != null)
                     {
                         // SyncFlush will Commit()
-                        if (CommittingBatch != null)
-                        {
-                            CommittingBatch(Channel, new EventArgs());
-                        }
+                        CommittingBatch?.Invoke(Channel, new EventArgs());
```
Comment (ThomasGoulet73) [inline src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/MediaContext.cs:999]: This is more of a style change, I don't think this does anything for race conditions. It generates the same code with a manual null check or a compiler-generated null check: https://sharplab.io/#v2:D4AQTAjAsAUCDMACciDCiDetE+UkALIgLIQAUAlNrljLvYgJYBmiZqA9gLZeMAufRgDsA5gCEAhnwDGAC0QBCALyIhAVwA2GqnQY5aevZx79BoyTNll1WgDSrN2gNzU9AX1eIPunJ4TIiYjBKTwNDY14BYXEpOQB+ADoASSEANw4AawBTa0d7G2dPb3o/JCzUrKE+RABRCqqACQkhABMNLIAnNG5IsxjLFxg3IA=

Comment (brunom) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/MediaContext.cs:999]: @ThomasGoulet73 Not the same code. M2 copies the event and is then immune to further changes.

Comment (ThomasGoulet73) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/MediaContext.cs:999]: Turns out you're right! I made the wrong assumption that invoking CommittingBatch was simply using it as a reference but turns out it does make a copy.

Here's the code I used to test this (It simulates a race condition):
https://sharplab.io/#v2:C4LgTgrgdgPgAgJgAwFgBQiCM65IARyYCsA3OugMJ4DGeAvHlAKYDueFAFAJRlrrUA6CgHsAtqICWwYBKgBzAEIBDYNQAWeANQMAogDcmUYJl44AbHmD0aAgCpMAzse6mMATgJmBAESYAbJQBPDkwAfSQInnI0QRFxKRl5ZVUNAFpdAyMTaLgPYFc4ABY8fUNnYQAjACsmaisHQwATJjAAGhLM4ABBMDkHPCVehy50AG90AF8cgGYCBHYx9Dxlglm4AA5PPHsnTG4llfG0FZOOsoAJJShGvxaaMUlpWUUVdWtgNQkHIQeE5+T1K5TitCG4OAASABEADkIH4/CA8KNqL8nklXho6AwoHC/BNIVE+MdgQQPHAvL4AsEEOFIkCSaCITDcYjkajEi8UvRsbj8YSTlMiSc4GtNuTto5gAh9sTlkcSQRMGCobD4ay4o8OQDMTz4Xz6cDcp4fP4ghwaREkISDoalUzVQikRq/uiuVjGLyCbwBdFhbMmJ0zkZLtdbmB2Oz/hjeBMgA==

C.Test1 prints
```
Null: False
Null: False
```

C.Test2 prints
```
Null: False
Null: True
```

**Pull request merged**